Tests/Unit/Registry/ContainerRepositoryRegistryTest.php

```diff
@@ -20,35 +20,59 @@ class ContainerRepositoryRegistryTest extends ProphecyTestCase
     public function setUp()
     {
         $this->container = $this->prophesize('Symfony\Component\DependencyInjection\ContainerInterface');
-        $this->repo1 = $this->prophesize('Puli\Repository\Api\ResourceRepository');
-        $this->repo2 = $this->prophesize('Puli\Repository\Api\ResourceRepository');
         $serviceMap = array(
             'one' => 'repo1_service',
             'two' => 'repo2_service',
         );
 
-        $containerMap = array(
-            'repo1_service' => $this->repo1->reveal(),
-            'repo2_service' => $this->repo2->reveal(),
-        );
-        $this->container->get(Argument::any())->will(function ($args) use ($containerMap) {
-            $name = array_shift($args);
-            return $containerMap[$name];
-        });
-
         $this->registry = new ContainerRepositoryRegistry($this->container->reveal(), $serviceMap);
     }
 
     public function testGet()
     {
-        $res = $this->registry->get('one');
-        $this->assertSame($res, $this->repo1->reveal());
+        $this->container->get('repo1_service')->shouldBeCalled();
+
+        $this->registry->get('one');
+    }
+
+    /**
+     * @dataProvider getGetDefaultRepositoryData
+     */
+    public function testGetDefaultRepository($defaultId = false, $shouldBeCalled = true)
+    {
+        if ($shouldBeCalled) {
+            $this->container
+                ->get(false === $defaultId ? 'cmf_resource.repository.default' : $defaultId)
+                ->shouldBeCalled();
+        }
+
+        if (false !== $defaultId) {
+            $registry = new ContainerRepositoryRegistry($this->container->reveal(), array(), array(), 'the_default');
+        } else {
+            $registry = new ContainerRepositoryRegistry($this->container->reveal());
+        }
+
+        $registry->get();
+    }
+
+    public function getGetDefaultRepositoryData()
```

Comment:
I always put the provider before the test, and call it `provideGetDefaultRepository`